contrib/cirrus/integration_test.sh

```diff
@@ -5,6 +5,14 @@ source $(dirname $0)/lib.sh
 
 req_env_var GOSRC SCRIPT_BASE OS_RELEASE_ID OS_RELEASE_VER CONTAINER_RUNTIME
 
+# Our name must be of the form xxxx_test or xxxx_test.sh, where xxxx is
+# the test suite to run; currently (2019-05) the only option is 'integration'
+# but pr2947 intends to add 'system'.
+TESTSUITE=$(expr $(basename $0) : '\(.*\)_test')
+if [[ -z $TESTSUITE ]]; then
+    die 1 "Script name is not of the form xxxx_test.sh"
+fi
+
```

Comment:
Could an env. var be used for the suite selection instead?  Primary reason being that env. var. based matrices are simple to setup and understand in Cirrus-CI (it's not possible to directly matrix over `foo_script: /blah/foobar.sh`).  It's desirable for the whole of CI to be maintainable by bash lay-persons - "Ask Ed/Chris" simply doesn't scale well :smile:  So hopefully you can understand why I'm leery about adding another operational mechanism (vs. env. vars).  

One idea could be to rework the uses of `$SPECIALMODE` into `$TESTSUITE`, and have this script use that to call specialized dedicated scripts for the suite and local-vs-remote mode.  I'm not married to the way it works today - rearranging it simpler is a completely viable option.

Another idea could be to combine the setup, unit, integration, and rootless scripts. Similar to (or perhaps along with) what Brent did in `container_test.sh`, using command-line flags to control operations.  This is probably more involved work, but would cut down a TON on the duplication we have today.

Lastly, I'm _okay_ with this as written, and addressing the complexity via another PR.  Just wanted to bring the issue into the light, in case there's an easy solution while you're into these files anyway.  We can only go so long with peace-meal feature addition before the whole thing starts swaying in the wind :smile:

Reply:
>Could an env. var be used for the suite selection instead?

I imagine so, but I've been really struggling to understand what that gains us. It seems to be inconsistent with your feedback in #2947:

> My preference would be to have make localsystem run in a separate (for example) systemtest_script (in .cirrus.yml).

That's how I was trying to address it, by making small changes to an existing script such that, in a subsequent PR, I could add a simple `system_test_script:` to `testing_task`. This seemed like the simplest and most maintainable way to get my BATS tests to run. What have I misunderstood?

Reply:
Ahh okay I gotcha now.  What you have here is fine for now...

Reply:
> This is another thing I don't understand. There's a matrix in .cirrus.yml that specifies TEST_REMOTE_CLIENT: true/false, but it seems to be unrelated to the matrix (in special_testing_task) that defines SPECIALMODE. Every time I try to dive into that I get a headache, most recently right now, so I think I'll have to try again in the morning...

...yeah this is what I was getting at, it gives me a headache as well.  Brent and I talked it over, and he's working a PR somewhere that flattens out the matrix stuffs a bit.  Anyway, now we both have headaches, so maybe given time, inspiration for a more elegant arrangement will come to us.

As for including system-tests into PR/Branch testing, the original concern (someone, can't remember) brought to me was, it would make testing run longer.  TBH, I would *love* to have that problem to solve (meaning, a team-effort in beefing up the system-tests).  For now I think they're fast enough, getting them run-time and lots of daylight isn't a bad thing at all.